scripts/get-sources

```diff
@@ -43,102 +42,104 @@ source "$BUILDER_DIR/scripts/common"
 
 url_var="GIT_URL_${COMPONENT//-/_}"
 
-if [ -n "$GIT_URL" ]; then
+if [ -n "${GIT_URL-}" ]; then
     GIT_URL="$GIT_URL"
-elif [ -n "${!url_var}" ]; then
+elif [ -n "${!url_var-}" ]; then
     GIT_URL="${!url_var}"
 else
     GIT_URL=$GIT_BASEURL/$GIT_PREFIX$COMPONENT$GIT_SUFFIX
 fi
 
 # Override GIT_URL with GIT_REMOTE if given
-[ -n "$GIT_REMOTE" ] && GIT_URL=$GIT_REMOTE
+[ -n "${GIT_REMOTE-}" ] && GIT_URL=$GIT_REMOTE
 
 branch_var="BRANCH_${COMPONENT//-/_}"
 
-if [ -n "${!branch_var}" ]; then
+if [ -n "${!branch_var-}" ]; then
     BRANCH="${!branch_var}"
 fi
 
+: "${IGNORE_MISSING=0}"
+
 GIT_OPTIONS=
 if [ "$GIT_CLONE_FAST" = "1" ]; then
     GIT_OPTIONS+="--depth=1"
 fi
 
-fresh_clone=0
-if [ "$REPO" == "." ] || [ -d "$REPO" ] && [ "$CLEAN" != '1' ]; then
-    cd $REPO
-    if [ "$(git rev-parse --is-shallow-repository)" == "true" ] && [ "$GIT_CLONE_FAST" != "1" ]; then
+if ! [[ "$BRANCH" =~ ^[A-Za-z][A-Za-z0-9._-]+$ ]]; then
+    printf 'Invalid branch %q\n' "$BRANCH"
+    exit 1
+elif ! { [[ "$REPO" = '.' ]] || [[ "$REPO" =~ ^qubes-src/[A-Za-z][A-Za-z0-9-]*$ ]]; }; then
+    printf 'Invalid repository %q\n' "$REPO"
+    exit 1
+fi
+fresh_clone=false
+if [ "$REPO" == "." ] || [ -d "$REPO" ] && [ "${CLEAN-}" != '1' ]; then
+    cd "$REPO"
+    if [ "$GIT_CLONE_FAST" != "1" ] && [ "$(git rev-parse --is-shallow-repository)" == "true" ]; then
         GIT_OPTIONS+="--unshallow"
     fi
     print_headers
-    if ! git fetch $GIT_OPTIONS -q "$GIT_URL" --tags $BRANCH; then
+    if ! git fetch $GIT_OPTIONS -q "$GIT_URL" --tags "$BRANCH"; then
         if [ "$IGNORE_MISSING" == "1" ]; then exit 0; else exit 1; fi
     fi
-    VERIFY_REF=FETCH_HEAD
+    VERIFY_REF=FETCH_HEAD fresh_clone=false
     # shellcheck disable=SC2103
     cd - >/dev/null
 else
-    rm -rf $REPO
+    rm -rf "$REPO"
     if [ "$(git rev-parse --is-shallow-repository)" == "true" ] && [ "$GIT_CLONE_FAST" != "1" ]; then
         GIT_OPTIONS+="--unshallow"
     fi
     print_headers
-    if ! git clone $GIT_OPTIONS -n -q -b $BRANCH "$GIT_URL" $REPO; then
+    if ! git clone $GIT_OPTIONS -n -q -b $BRANCH "$GIT_URL" "$REPO"; then
         if [ "$IGNORE_MISSING" == "1" ]; then exit 0; else exit 1; fi
     fi
-    VERIFY_REF=HEAD
-    fresh_clone=1
+    VERIFY_REF=HEAD fresh_clone=:
 fi
 
-verify=true
-if [ "$NO_CHECK" == "1" ] || elementIn "$COMPONENT" ${NO_CHECK[@]}; then
-    echo "--> $COMPONENT has NO_CHECK enabled"
-    echo "--> NOT Verifying tags..."
-    verify=false
-fi
```

Comment:
Would it be okay to require a different value for this?  I would prefer something like `insecure-no-check-tags`.

Reply:
I had not considered that a tag implies authorization, but indeed it does.  What about replacing `NO_CHECK` with a multivalued `CHECK` option?

- `signed-tag` (default) checks that there is a signed tag.
- `signed-tag-or-commit` checks that either there is a signed tag, or that the commit itself is signed.
- `insecure-no-check` turns off all checking.
- anything else is an error

Reply:
Yes, that should be fine.

Reply:
I think there was a race condition in those comments. My "Yes, that should be fine." applied to "insecure-no-check-tags" value, not multi-value option. The latter wouldn't work because it won't be possible to specify to which components it shoul apply. But given what you've actually implemented, it seems you've figured it out too :)